### PR TITLE
k3s: Add k3s 1.24.3 and k3s 1.23.8

### PR DESCRIPTION
**Pull request by ag17sep**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?

This PR updates the k3s version to 1.24.3 and adds k3s version 1.23.8 to build 2 rpms. 

###### Change Log  

- Change
- Change
- Change

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Built and tested locally using the Mariner toolkit.
- BuddyBuild AMD64 : [230721](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=230721&view=results)
- BuddyBuild ARM64 : [230729](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=230729&view=results)

Closes: #3582

### CODE REVIEW

File: SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md [hunk 1]
```diff
@@ -15,5 +15,5 @@ The CBL-Mariner SPEC files originated from a variety of sources with varying lic
 | OpenMamba | [Openmamba GPLv2 License](https://www.gnu.org/licenses/old-licenses/gpl-2.0.txt) | bash-completion |
```
Comment (mfrw) [inline SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md:15]: closes:
- https://github.com/microsoft/CBL-Mariner/pull/3582

File: SPECS/k3s/k3s-1.23.8.signatures.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "Signatures": {
+     "k3s-1.23.8-vendor.tar.gz": "017636ddaf542442b9c8992fb8646a48a5d57efdecb4861c626e8ad919bbf905",
```
Comment (mfrw) [inline SPECS/k3s/k3s-1.23.8.signatures.json:3]: This source does not seem to be available on the `blob-store`.
Can we please double check.

Comment (ag17sep) [reply SPECS/k3s/k3s-1.23.8.signatures.json:3]: It should be available now

Comment (mfrw) [reply SPECS/k3s/k3s-1.23.8.signatures.json:3]: Sure, the signature does not seem to match to the one in the `blob-store` though.
I am getting: `3c96e864d0e89e318ecdd62e1750f787d1b622feeb240f7b86d9f3280447aeda` - Can you please double check again :)

Comment (ag17sep) [reply SPECS/k3s/k3s-1.23.8.signatures.json]: Updated. Thanks for pointing that out :)

File: SPECS/k3s/k3s.spec [hunk 3]
```diff
@@ -83,9 +82,10 @@ exit 0
 %{install_sh}
 
 %changelog
+* Tue Aug 30 2022 Animesh Garg <animeshgarg@microsoft.com> 1.24.3-1
+- Updated k3s to 1.24.3
```
Comment (mfrw) [inline SPECS/k3s/k3s.spec:86]:
```suggestion
- Updated k3s to 1.24.3

```

File: SPECS/k3s/k3s.spec [hunk 4]
```diff
@@ -83,9 +82,10 @@ exit 0
 %{install_sh}
 
 %changelog
+* Tue Aug 30 2022 Animesh Garg <animeshgarg@microsoft.com> 1.24.3-1
```
Comment (mfrw) [inline SPECS/k3s/k3s.spec:85]:
```suggestion
* Tue Aug 30 2022 Animesh Garg <animeshgarg@microsoft.com> - 1.24.3-1
```

**Review by oliviacrain — APPROVED** [review]

**Comment by oliviacrain** [comment]
LGTM. Please close #3468 and #3582 once merged, since it looks like this PR supersedes them.

**Review by anphel31 — APPROVED** [review]

**Review by mfrw — APPROVED** [review]
LGTM :)

**Comment by lilustga** [comment]
Can this be re-opened? It's not made redundant by PR #3582 because we require supporting multiple minor versions simultaneously.

**Comment by mfrw** [comment]
Do we want to revert this change ?
As this was merged.
This specific PR does support multiple versions.

**Comment by lilustga** [comment]
No need to revert the change, I didn't realize this PR included the changes from that one. Thanks!

**Pull request merged**